src/auth/Crypto.cc

```diff
@@ -353,19 +365,405 @@ CryptoKeyHandler *CryptoAES::get_key_handler(const bufferptr& secret,
 {
   CryptoAESKeyHandler *ckh = new CryptoAESKeyHandler;
   ostringstream oss;
-  if (ckh->init(secret, oss) < 0) {
+  if (ckh->init(secret, bufferptr((unsigned)0), oss) < 0) {
```

Comment:
the [buffer::ptr::ptr(unsigned l)](https://github.com/aclamk/ceph/blob/248637b4090dfd91573d463462f5b7b83b46b9e4/src/common/buffer.cc#L780-L785) constructor will allocate a raw buffer even if length is 0. you probably want `bufferptr()` instead